src/World.cpp

```diff
@@ -691,9 +696,61 @@ void cWorld::GenerateRandomSpawn(void)
 		}
 	}  // for i - 100*
 
-	m_SpawnY = (double)GetHeight((int)m_SpawnX, (int)m_SpawnZ) + 1.6f;  // 1.6f to accomodate player height
+	if (foundSpawnPoint)
+	{
+		m_SpawnY = (double)GetHeight((int)m_SpawnX, (int)m_SpawnZ) + 1.6f;  // 1.6f to accomodate player height
+
+		LOGINFO("Generated random spawnpoint position at {%i, %i, %i}", (int)m_SpawnX, (int)m_SpawnY, (int)m_SpawnZ);
+	}
+	else
+	{
+		LOGINFO("Did not find an acceptable spawnpoint. Generated a random spawnpoint position at {%i, %i, %i}", (int)m_SpawnX, (int)m_SpawnY, (int)m_SpawnZ);
+	}  // Maybe widen the search instead?
+
+}
+
+
+
+
 
-	LOGINFO("Generated random spawnpoint position {%i, %i, %i}", (int)m_SpawnX, (int)m_SpawnY, (int)m_SpawnZ);
+bool cWorld::CheckPlayerSpawnPoint(int a_PosX, int a_PosY, int a_PosZ)
+{
+	static const struct
+	{
+		int x, z;
+	} Coords[] =
+	{
+		{ 0, 0 },
+		{ -1, 0 },
+		{ 1, 0 },
+		{ 0, -1 },
+		{ 0, 1 },
+	};
+	for (size_t i = 0; i < ARRAYCOUNT(Coords); i++)
+	{
+		BLOCKTYPE BlockType = GetBlock(a_PosX + Coords[i].x, a_PosY, a_PosZ + Coords[i].x);
+
+		if (cBlockInfo::IsSolid(BlockType) && IsBlockLiquid(BlockType))
+		{
+			return false;
+		}
+	}  // for i - Coords[]
+	// Check if block below is solid
+	BLOCKTYPE BlockType = GetBlock(a_PosX, a_PosY - 1, a_PosZ);
+	if (!cBlockInfo::IsSolid(BlockType))
+	{
+		return false;
+	}
+	// Players should spawn with only air above.
```

Comment:
Add an empty line above each of these comments, and add a comment for the first code block.